### PR TITLE
refactor: use structopt for CLI root

### PR DESCRIPTION
**Pull request by tustvold**
Bit of of a yak-shaving PR but wanted to get some familiarity with structopt + clap.

By using structopt this unifies the definition and parsing logic, which I at least think makes the code more legible and avoids some duplication.

It also renames some of the modules under `src/commands` to better reflect their purpose

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -57,170 +62,132 @@ Examples:
 
     # Dumps storage statistics about out.parquet to stdout
     influxdb_iox stats out.parquet
-"#;
-    // load all environment variables from .env before doing anything
-    load_dotenv();
+"#
+)]
+struct Opt {
```
Comment (alamb) [inline src/main.rs:67]: `struct Opt` with `structops` 🤔 

What about calling this structure `Config` for consistency with the other modules?

File: src/main.rs [hunk 2]
```diff
@@ -57,170 +62,132 @@ Examples:
...
 
-    let tokio_runtime = get_runtime(matches.value_of("num-threads"))?;
-    tokio_runtime.block_on(dispatch_args(matches));
+    #[structopt(subcommand)]
+    command: Option<Command>,
+}
 
-    info!("InfluxDB IOx server shutting down");
-    Ok(())
+#[derive(Debug, StructOpt)]
```
Comment (alamb) [inline src/main.rs:80]: This is cool

**Review by alamb — APPROVED** [review]
I think this is a great cleanup and improvement. Nice work @tustvold 👍

**Review by carols10cents — APPROVED** [review]
LGTM! Nice cleanup! 🧹

**Pull request merged**